tests/integration/states/test_x509.py

```diff
@@ -19,48 +22,54 @@
+        with salt.utils.files.fopen(cert_path) as fp:
+            cls.x509_cert_text = fp.read()
+
+    def run_function(self, *args, **kwargs):
+        ret = super(x509Test, self).run_function(*args, **kwargs)
+        log.debug('ret = %s', ret)
+        return ret
 
-    @staticmethod
-    def get_cert_lines(path):
-        lines = []
-        started = False
-        with salt.utils.files.fopen(path, 'rb') as fp:
-            for line in fp:
-                if line.find(b'-----BEGIN CERTIFICATE-----') != -1:
-                    started = True
-                    continue
-                if line.find(b'-----END CERTIFICATE-----') != -1:
-                    break
-                if started:
-                    lines.append(line.strip())
-        return lines
+    @with_tempfile(suffix='.pem', create=False)
+    def test_issue_49027(self, pemfile):
+        ret = self.run_state(
+            'x509.pem_managed',
+            name=pemfile,
+            text=self.x509_cert_text)
+        assert isinstance(ret, dict), ret
+        ret = ret[next(iter(ret))]
+        assert ret.get('result') is True, ret
+        with salt.utils.files.fopen(pemfile) as fp:
+            result = fp.readlines()
+        self.assertEqual(self.x509_cert_text.splitlines(True), result)
 
-    @skipIf(not HAS_M2CRYPTO, 'Skip when no M2Crypto found')
-    def test_issue_49027(self):
-        expected = self.get_cert_lines(os.path.join(BASE_FILES, 'issue-49027.sls'))
-        started = False
-        ret = self.run_function('state.sls', ['issue-49027'])
-        log.warn("ret = %s", repr(ret))
-        self.assertSaltTrueReturn(ret)
-        self.assertEqual(expected, self.get_cert_lines('/test-49027.crt'))
+    @with_tempfile(suffix='.crt', create=False)
+    @with_tempfile(suffix='.key', create=False)
+    def test_issue_49008(self, keyfile, crtfile):
+        ret = self.run_function(
+            'state.apply',
+            ['issue-49008'],
+            pillar={'keyfile': keyfile, 'crtfile': crtfile})
+        assert isinstance(ret, dict), ret
+        for state_result in six.itervalues(ret):
+            assert state_result['result'] is True, state_result
+        assert os.path.exists(keyfile)
+        assert os.path.exists(crtfile)
 
-    @skipIf(not HAS_M2CRYPTO, 'Skip when no M2Crypto found')
-    def test_issue_49008(self):
-        ret = self.run_function('state.sls', ['issue-49008'])
-        log.warn("ret = %s", repr(ret))
-        self.assertSaltTrueReturn(ret)
-        self.assertTrue(os.path.exists('/test-ca-49008.key'))
-        self.assertTrue(os.path.exists('/test-ca-49008.crt'))
+    @with_tempfile(suffix='.pem', create=False)
+    def test_issue_50637(self, pemfile):
+        ret = self.run_state(
+            'x509.pem_managed',
+            name=pemfile,
+            text=self.x509_cert_text,
+            show_changes=False,
+            encoding='utf-8',
+            encoding_errors='ignore',
+            mode=640)
+        print ret
```

Comment:
Left over right?

Reply:
Also, missing asserts?

Reply:
It turns out that incomplete test was something I was working on for an issue that was a duplicate of something already being tested. So I have removed it.